src/prefect/utilities/feature_flags.py

```diff
@@ -0,0 +1,171 @@
+"""
+This module contains code related to feature flagging.
+
+For more guidance, read the [Feature Flags](/contributing/feature_flags/) documentation.
+"""
+import threading
+from typing import Any, Iterable, List, Optional
+
+from flipper import Condition, FeatureFlagClient, MemoryFeatureFlagStore
+from flipper.bucketing.base import AbstractBucketer
+from flipper.flag import FeatureFlag
+
+from prefect import settings
+
+_client: Optional[FeatureFlagClient] = None
+_in_memory_store: Optional[MemoryFeatureFlagStore] = None
+
+
+def get_feature_flag_client() -> FeatureFlagClient:
+    global _client, _in_memory_store
+
+    if _client:
+        return _client
+
+    with threading.Lock():
+        _in_memory_store = MemoryFeatureFlagStore()
+        _client = FeatureFlagClient(_in_memory_store)
+
+    return _client
+
+
+def create_if_missing(
+    flag_name: str,
+    is_enabled: bool = False,
+    client_data: Optional[dict] = None,
+    bucketer: Optional[AbstractBucketer] = None,
+    conditions: Optional[Iterable[Condition]] = None,
+    client: FeatureFlagClient = None,
+) -> Optional[FeatureFlag]:
+    """
+    Create a feature flag if a flag matching the given name does not
+    already exist.
+
+    Args:
+        flag_name: The name of the feature flag.
+        is_enabled: the initial enabled/disabled state of the flag if
+                    this function creates it
+        client_data: arbitrary data that we should store with the flag
+        bucketer: an optional "bucketter" from the `flipper` module, e.g.
+                  `PercentageBucketer`, to use when determining if the
+                  flag is enabled
+        conditions: an optional iterable of `Condition` instances from the
```

Comment:
@tpdorsey Want to check out this docstring again? There are some difficult-communicate subtleties here that the type annotations explain better (e.g., `AbstractBucketer` is an interface, but `Condition` is an instance).